src/main/java/ch/myniva/gradle/caching/s3/internal/AwsS3BuildCacheServiceFactory.java

```diff
@@ -71,17 +71,18 @@ private void verifyConfig(AwsS3BuildCache config) {
   private AmazonS3 createS3Client(AwsS3BuildCache config) {
     AmazonS3 s3;
     try {
+      AmazonS3ClientBuilder s3Builder = AmazonS3ClientBuilder.standard();
+      if (config.getAwsAccessKeyId() != null && config.getAwsSecretKey() != null) {
```

Comment:
I suggest using `isNullOrEmpty` instead.

Reply:
done, but I wonder if it won't be better to use `hasValue` in that case